bson/tests/test_objectid.py

```diff
@@ -511,18 +510,19 @@ def test_pid(self):
         self.assertTrue(oid_generated_on_client(ObjectId()))
 
     def test_generation_time(self):
-        d1 = datetime.datetime.utcnow()
+        d1 = datetime.datetime.now(datetime.UTC)
         d2 = ObjectId().generation_time
 
         self.assertEqual(utc, d2.tzinfo)
-        d2 = d2.replace(tzinfo=None)
         self.assertTrue(d2 - d1 < datetime.timedelta(seconds=2))
 
     def test_from_datetime(self):
         if 'PyPy 1.8.0' in sys.version:
             # See https://bugs.pypy.org/issue1092
             raise SkipTest("datetime.timedelta is broken in pypy 1.8.0")
-        d = datetime.datetime.utcnow()
+        # note: could use tz aware datetimes here?
+        d = datetime.datetime.now(datetime.UTC).replace(tzinfo=None)
```

Comment:
Py3.9?

Reply:
Hmm -- 3.9 is not so far from end of life -- time to dump it now? the last version will keep working .....

Reply:
https://endoflife.date/python - the security support ends in 10 months, so it is maintained in a way.

Reply:
> the security support ends in 10 months, so it is maintained in a way.

yes it's not end of life yet -- but will be in 10 months, so is it worth supporting in the next version of this package? your call :-).

Reply:
turns out there was very little to do to keep it py3.9 compatible, so I've done that.

Reply:
> And do you need to compute now with UTC and then replacing the latter with None?

well, I agree, but that's the new way to do it -- the naive utcnow() is deprecated -- I don't think it should be -- a naive utc datetime is useful, but there you go.

I figure while updating, now is the time to address deprecations.